### ISSUE TITLE
Repository and data structure

### ISSUE THREAD

**Issue by ebuhle** [ebuhle/chinook-adult-age#1]
OK @JLGnotes, here is a dump of my old R directory for this project. I'm embarrassed by how badly organized it was, and how messy the code itself is, but on the upside I've come a long way in the last seven years.

All of the previous files are in subdirectories of `/old`, but I intend for new work to live in the respective top-level directories. Most immediately useful will be the main data set `/old/data/srssW.age.txt`, the main analysis script `/old/src/adult_age_PIT_analysis.R`, and the JAGS model `/old/src/HB_OP_adult_age_PIT.R` (for **h**ierarchical **B**ayesian **o**rdinal **p**robit). Note that the scripts will not run as written because I previously had everything in one top-level directory :flushed:, so they would need to be updated with **here** to be usable. The data structure looks like so:

```
> head(srssW.age)
         tag age year LGR.smolt.date LGR.smolt.day route rear.type length LGR.flow
1 510005030A   2 1998     1998-04-07             7     I         W    105     68.7
2 5105721C62   3 1998     1998-04-07             7     I         W    110     68.7
3 51057C462B   3 1998     1998-04-07             7     I         W    105     68.7
4 510A4E3755   3 1998     1998-04-07             7     I         W    103     68.7
5 510A511712   2 1998     1998-04-07             7     I         W    107     68.7
6 510C0A3A6C   2 1998     1998-04-07             7     I         W    116     68.7
  LGR.temp      CUI CUMCUI NPGO ONI  PNI   PDO
1     8.78 29.20879 -10064 0.52 0.9 0.33 -0.07
2     8.78 29.20879 -10064 0.52 0.9 0.33 -0.07
3     8.78 29.20879 -10064 0.52 0.9 0.33 -0.07
4     8.78 29.20879 -10064 0.52 0.9 0.33 -0.07
5     8.78 29.20879 -10064 0.52 0.9 0.33 -0.07
6     8.78 29.20879 -10064 0.52 0.9 0.33 -0.07
```

**Comment by JLGnotes** [comment]
@ebuhle Thank you very much for the dump of files. Much appreciated. 

Thanks for the comment about needing to update file paths with the `here` function. But my guess is that what you have from 7 years ago is still more organized than what I generally have! 

Thanks as well for the data and structure. I'll ask our database manager to update this data set. Although this reminds me that our 'DART Transport/In-river' filter on the PIT data (https://www.cbr.washington.edu/dart/metadata/pit#transport) and 'DART Life Stage' filter (https://www.cbr.washington.edu/dart/metadata/pit#lifestage) might need updating. Any bugs in these filters would need to be corrected before we finalize our analysis and results. Our database manager will be working on that soon, and it'll be good to get him connected with Ben. Hopefully, in the meantime, the preliminary data set, updated with the last 7 years will be sufficient to move forward on analysis. 

I was about to say that I'll think some more about why the results differ between the data that includes trap data vs LGR only data, but perhaps we should wait until we get the 1998-2019 data. The results with 1998-2019 data may be quite different from what we saw with the 1998-2012 data. Perhaps, we'll see the same patterns as Kevin Malone in both the trap-included data and the LGR-only data when it includes recent years of data. 

On a different note, starting Monday, we can invite Caitlin O'Brien (csobrien@uw.edu) to this repo.

**Comment by ebuhle** [comment]
Cool. Not sure I understand the implications of needing to debug those database filters, but it sounds like that won't hold us back from updating the data set and getting started on the analysis.

> On a different note, starting Monday, we can invite Caitlin O'Brien to this repo.

Great, but of course I'll need her GH username not her email in order to do that. I took the liberty of redacting her email address from your post so it's not sitting out there in public.

**Comment by JLGnotes** [comment]
The filters might affect what PIT data get included in the data set and how we designate each fish being transported or in-river. 

Oh yeah, forgot this is a public repo and not a private one. btw, I'm not sure if she has a GitHub account, hence me adding her email.

**Comment by ebuhle** [comment]
> The filters might affect what PIT data get included in the data set and how we designate each fish being transported or in-river.

Sure, got that; I guess I meant more specific implications. :wink: It sounded like maybe you saw something concerning in the data subset I posted, but probably I just need coffee.

And yes, let me know if / when Caitlin is on here. Relatedly, I have a sneaking suspicion that I will end up puttering around with initial attempts to refactor the code for this analysis, in which case we would have a firmer jumping-off point to move forward.

**Comment by JLGnotes** [comment]
> It sounded like maybe you saw something concerning in the data subset I posted, but probably I just need coffee.

No, just something more on our end of processes. There may be no effect in the end. Good to make sure. 

> I have a sneaking suspicion that I will end up puttering around with initial attempts to refactor the code for this analysis, in which case we would have a firmer jumping-off point to move forward.

Thank you in advance, Eric. Much appreciated.

**Comment by JLGnotes** [comment]
It turns out that the data we need can be subset from the data I use for the CJS-individual traits analysis (aka Ecosphere study). 

I'm just working on subsetting the data now and was wondering if we want juvenile detections just at Lower Granite or also at other dams downstream of LWG. Similarly, do we want adult detections above Bonneville? I'm going to keep it simple for now, with just having first and last juvenile detections at LWG and first and last adult detections at BON. But I can add more columns for these other detection sites, if you think it's needed.

**Comment by ebuhle** [comment]
Ben is more qualified to address this than I am, but here are some thoughts. 

Re additional downstream detections, it seems like this would create more headaches w.r.t. covariates. Where would river conditions be measured for these fish, given that their passage date at LGR is unknown? And the length imputation for fish tagged as summer parr would get even trickier, since downstream growth rates surely differ from parr-to-smolt growth rates. How much would this increase the sample size of returning adults?

Re additional upstream detections, this seems like a good idea since we're only interested in adult age, but maybe there are nuances I'm not appreciating. Detection at BON is generally quite high, though, no? So this might not increase our sample size by that much?

**Comment by JLGnotes** [comment]
> Re additional downstream detections, it seems like this would create more headaches w.r.t. covariates. Where would river conditions be measured for these fish, given that their passage date at LGR is unknown? And the length imputation for fish tagged as summer parr would get even trickier, since downstream growth rates surely differ from parr-to-smolt growth rates. How much would this increase the sample size of returning adults?

I agree that including these data would create more headaches, and not worth delving into. These would be downstream detections in addition to detections at LGR. Not fish that were not detected at LGR. I was thinking that these detections could be used to determine number of bypasses experienced, and could be a covariate. But I don't quite see how this could affect age at maturation. 


> Re additional upstream detections, this seems like a good idea since we're only interested in adult age, but maybe there are nuances I'm not appreciating. Detection at BON is generally quite high, though, no? So this might not increase our sample size by that much?

Yes, BON detection is quite high. But with BON-only detections, we see 1195 adults. And with BON, MCN, ICH, and LGR detections, we see 1320 adults. 

In case you noticed the column 'growthrate', I just asked our database manager where that came from. It is:
(average recapture length – release length)/(average (recapture time – tag time)
Please ignore this column, and any updated data sets I do, I'll exclude this column. 
We're already dealing with much hesitation in using individually predicted length from a growth model. I don't think we'd use these 'growthrate' estimates.

**Comment by ebuhle** [comment]
> These would be downstream detections in addition to detections at LGR. Not fish that were not detected at LGR. I was thinking that these detections could be used to determine number of bypasses experienced, and could be a covariate. But I don't quite see how this could affect age at maturation.

Ah, gotcha. Yes, I'm not sure if this is a relevant covariate. It's also subject to measurement error because of imperfect detection, so wouldn't we have to model it (i.e., with CJS) to estimate the most likely number of bypasses?

**Comment by JLGnotes** [comment]
Yes, good point. I had a momentary relapse of "reluctance to throw data out" but not for good reason.

**Comment by ebuhle** [comment]
BTW, we should really get Ben on here.

**Comment by bensandford** [comment]
Hey, guess what!  I'm still "in".  I guess I got dropped from gitlab??  I'm posting the comments I sent via email here:

Anyway, just to provide a quick comment on the "use more than LGR for juveniles" and "use more than BON for adults".  The juvenile side I would just stick with LGR - sometimes transport studies use transport from all 3 sites, but that always makes me leery since it seems hard to pair with returned-to-river fish at each of them as well.  Also, the idea of where to measure covariates makes the case for LGR-only logical in my opinion.  On the adult side, I would use all dams - certainly BON should have a large percentage, but Jenn's numbers indicate it's certainly not 100% and all first-time detections above had to pass BON so should be included - further, we don't care when they passed BON as long as it was in the same year, right?  I think we should examine the "not detected fish at BON" to see if there are reasons for those misses (eg periods of non-detection or just random).

Good to read your discusson :)

I'd like to read up on the methods/implementation we are going to use, so happy to piggyback on reading you get for Caitlin.

Glad to make your acquaintance, Caitlin, "virtually textually" anyway :)

Ben

**Comment by JLGnotes** [comment]
I've updated the data set with adult detections upstream of BON. I've also excluded juveniles that were detected at LGR, returned in-river, and transported from downstream dams (i.e., LGS, LMN, MCN). So, now we're at N = 1224. 

Most of the adults that were detected upstream of BON, but not at BON, were detected at LGR. 

(Sorry for using LWG and LGR interchangeably. Trying to match what others use, but "others" as a whole use LWG and LGR!)

**Comment by bensandford** [comment]
The “adults detected not at BON but then at LGR” should all be from “way back” before good/any adult detection elsewhere and much less than full coverage at BON  - like before 2001 or so.Sent from my iPhoneOn Dec 16, 2022, at 4:45 PM, JLGnotes ***@***.***> wrote:﻿
I've updated the data set with adult detections upstream of BON. I've also excluded juveniles that were detected at LGR, returned in-river, and transported from downstream dams (i.e., LGS, LMN, MCN). So, now we're at N = 1224.
Most of the adults that were detected upstream of BON, but not at BON, were detected at LGR.
(Sorry for using LWG and LGR interchangeably. Trying to match what others use, but "others" as a whole use LWG and LGR!)

**Comment by JLGnotes** [comment]
Hey all,

So something funky happened and somehow there were only juveniles tagged above LGR and detected at LGR (i.e., not including juveniles tagged and released at LGR)... so we had a lot less adults than there actually are in the data set. Not sure how I missed the fact that there were so few adults across all those years; this should have been a red flag to me. 

Matt has updated the data set and Caitlin's been working with the new data. I think we can get rid of the temporary dataset based on the Dsynth data set now, and continue working with the new data set. 

Also, the Lower-Snake-Tucannon designation refers to the HUC. So these fish are pretty much all the fish at LGR, and not from a Tucannon site.
